### PR TITLE
MTD geometry: make dd4hep test independent on dd4hep length unit change

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

This PR replaces ```convertCmToMm``` with ```/dd4hep::mm``` in order to make the dd4hep MTD ideal geometry test code insensitive to changes in the DD4hep baseline unit of measurements for length, as done for many other subsystems. #32488 takes care of the RECO level geometry update.

#### PR validation:

The unit test shows an identical description for both DDD and DD4hep.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

File: Geometry/MTDCommonData/test/DD4hep_TestMTDIdealGeometry.cc [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 #include "DataFormats/Math/interface/GeantUnits.h"
```
Comment (cvuosalo) [inline Geometry/MTDCommonData/test/DD4hep_TestMTDIdealGeometry.cc:32]: Can line 32 be deleted?

Comment (cvuosalo) [reply Geometry/MTDCommonData/test/DD4hep_TestMTDIdealGeometry.cc:32]: Also, change `BuildFile.xml` to delete `DataFormats/Math` and add DD4hep?

Comment (fabiocos) [reply Geometry/MTDCommonData/test/DD4hep_TestMTDIdealGeometry.cc]: ```DataFormats/Math/interface/GeantUnits.h``` in itself is no more strictly needed, but ```DataFormats/Math/interface/angle_units.h``` that was included in it is still needed (and therefore it should not be removed from the BuildFile)

**Comment by fabiocos** [comment]
please test

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by qliphy** [comment]
+1

**Pull request merged**